### DISCUSSION TITLE
Why does an idle python service needs 1% to 1,6% of my CPU?

### DISCUSSION THREAD

**Discussion by AzureIP** [mhammond/pywin32#2606] [Q&A] [answered]
```
import win32serviceutil
import win32service
import cherrypy


class HelloWorld(object):

    @cherrypy.expose
    def index(self):
        return "Hello world!"


class MyService(win32serviceutil.ServiceFramework):
    _svc_name_ = "myName"
    _svc_display_name_ = "myName"

    def SvcDoRun(self):
        cherrypy.quickstart(HelloWorld())

    def SvcStop(self):
        self.ReportServiceStatus(win32service.SERVICE_STOP_PENDING)

        self.ReportServiceStatus(win32service.SERVICE_STOPPED)
        # very important for use with py2exe
        # otherwise the Service Controller never knows that it is stopped !


if __name__ == '__main__':
    win32serviceutil.HandleCommandLine(MyService)

```

Hi,

I created this simple service that contains a cherrypy webserver that provides a Hello World page. If I run this web server without a service it takes 0% CPU load. If I run this script above that runs the server as a service, it takes 1% to 1,6% of my CPU. A bit too mouch for doing nothing in my opinion. Why is that happening?

**Comment by ACrazyPencil** [comment]
What cpu do u have?

**Comment by ACrazyPencil** [comment]
and it’s reporting to windows

**Comment by ACrazyPencil** [comment, marked as answer]
not pywin32s problem but it’s cherrypys problem https://stackoverflow.com/questions/45888452/cherrypy-quickstart-consuming-high-cpu-usage-while-idle

**Comment by AzureIP** [comment]
I am not convinced. I run the same server as service and direct on the command line. And I notice the the cpu load only if I run it as a service.

**Reply by pizzapi2012** [reply]
ok.
